modules/service/src/main/resources/db/migration/V0980__flamingos2_longslit.sql

```diff
@@ -0,0 +1,164 @@
+-- and program combination across all observing modes.
+CREATE OR REPLACE VIEW v_observing_mode_group AS
+-- GMOS-N LongSlit
+  SELECT
+    m.c_mode_key,
+    o.c_program_id,
+    max(m.c_observation_id) as c_observation_id -- arbitrary, just pick one
+  FROM
+    t_gmos_north_long_slit m
+  LEFT JOIN t_observation o ON m.c_observation_id = o.c_observation_id
+  GROUP BY
+    m.c_mode_key,
+    o.c_program_id
+-- GMOS-S LongSlit
+UNION ALL
+  SELECT
+    m.c_mode_key,
+    o.c_program_id,
+    max(m.c_observation_id) as c_observation_id -- arbitrary, just pick one
+  FROM
+    t_gmos_south_long_slit m
+  LEFT JOIN t_observation o ON m.c_observation_id = o.c_observation_id
+  GROUP BY
+    m.c_mode_key,
+    o.c_program_id
+-- F2 LongSlit
+UNION ALL
+  SELECT
+    m.c_mode_key,
+    o.c_program_id,
+    max(m.c_observation_id) as c_observation_id -- arbitrary, just pick one
+  FROM
+    t_flamingos_2_long_slit m
+  LEFT JOIN t_observation o ON m.c_observation_id = o.c_observation_id
+  GROUP BY
+    m.c_mode_key,
+    o.c_program_id;
+
+-- Replace the observation view to add f2 long slit mode columns.
+DROP VIEW v_observation;
+
+CREATE OR REPLACE VIEW v_observation AS
+  SELECT o.*,
+  CASE WHEN o.c_explicit_ra              IS NOT NULL THEN o.c_observation_id END AS c_explicit_base_id,
+  CASE WHEN o.c_air_mass_min             IS NOT NULL THEN o.c_observation_id END AS c_air_mass_id,
+  CASE WHEN o.c_hour_angle_min           IS NOT NULL THEN o.c_observation_id END AS c_hour_angle_id,
+  CASE WHEN o.c_observing_mode_type      IS NOT NULL THEN o.c_observation_id END AS c_observing_mode_id,
+  CASE WHEN o.c_spec_wavelength          IS NOT NULL THEN o.c_observation_id END AS c_spec_wavelength_id,
+
+  CASE WHEN o.c_spec_exp_time_mode       IS NOT NULL THEN o.c_observation_id END AS c_spec_exp_time_mode_id,
+  CASE WHEN o.c_spec_exp_time_mode = 'signal_to_noise' THEN o.c_observation_id END AS c_spec_signal_to_noise_id,
+  CASE WHEN o.c_spec_exp_time_mode = 'time_and_count'  THEN o.c_observation_id END AS c_spec_time_and_count_id,
+
+  CASE WHEN o.c_spec_wavelength_coverage IS NOT NULL THEN o.c_observation_id END AS c_spec_wavelength_coverage_id,
+  CASE WHEN o.c_spec_focal_plane_angle   IS NOT NULL THEN o.c_observation_id END AS c_spec_focal_plane_angle_id,
+  CASE WHEN o.c_observation_duration     IS NOT NULL THEN o.c_observation_id END AS c_observation_duration_id,
+  c.c_active_start::timestamp + (c.c_active_end::timestamp - c.c_active_start::timestamp) * 0.5 AS c_reference_time
+  FROM t_observation o
+  LEFT JOIN t_proposal p on p.c_program_id = o.c_program_id
+  LEFT JOIN t_cfp c on p.c_cfp_id = c.c_cfp_id
+  LEFT JOIN t_gmos_north_long_slit  mode_gnls ON o.c_observation_id = mode_gnls.c_observation_id
+  LEFT JOIN t_gmos_south_long_slit  mode_gsls ON o.c_observation_id = mode_gsls.c_observation_id
+  LEFT JOIN t_flamingos_2_long_slit mode_f2ls ON o.c_observation_id = mode_f2ls.c_observation_id;
```

Comment:
I can't remember why we join on the modes there.  We don't seem to use it for anything?  I may have used them at one point, removed it, and forgot to clean up.  Perhaps we could remove these last three joins?  Am I missing something?

Reply:
Not sure, I took it from the existing table and assumed we needed it for f2 too
It could be problematic as we add more modes.

Reply:
It doesn't seem to contribute anything to the view so let's delete them?

Reply:
ok I'll update the table